hstream-io/HStream/IO/Worker.hs

```diff
@@ -176,7 +176,11 @@ recoverTask worker@Worker{..} name = do
     Nothing -> throwIO $ HE.ConnectorNotFound name
     Just (taskId, TaskMeta{taskInfoMeta=taskInfo@TaskInfo{..}}) -> do
       let newConnCfg = J.insert "hstream" (J.toJSON hsConfig) connectorConfig
-      createIOTaskFromTaskInfo worker taskId taskInfo{connectorConfig=newConnCfg} options True False False
+          newImage = if options.optFixedConnectorImage
+                     then taskConfig.tcImage
+                     else makeImage taskType taskTarget options
+          newTaskConfig = taskConfig{tcImage=newImage}
+      createIOTaskFromTaskInfo worker taskId taskInfo{connectorConfig=newConnCfg, taskConfig=newTaskConfig} options True False False
```

Comment:
If the connector is recovered with a new image, will the new configuration information be updated in the meta store? We need to make sure that subsequent visits to the meta store fetch the updated information, otherwise the relevant api (maybe describeConnector, listConnector, ...) will return incorrect results!

Reply:
No, the meta-store will not be updated, it is expected behavior, the image is something internal information, so describeConnector, listConnectors will not return this to users. What if admin users want to check a connector image? the only correct way is to inspect the docker container, the meta-store image means nothing.

Reply:
> What if admin users want to check a connector image? the only correct way is to inspect the docker container, the meta-store image means nothing.

I can't agree with it. If we have a lot of machines in our cluster, logging into each machine to check the image is a very inefficient task. There is always a need for a place to summarize this information for easy querying. Actually, `describe connector` give us this information https://github.com/hstreamdb/hstream/blob/main/hstream/src/HStream/Server/Handler/Admin.hs#L462

Also if we don't update the meta-store, there comes anthor problem. Supporse we are currently using image1, and I change the configuration to use image2 to recover the connector, and then turn off `fixed-connector-image` on the next reboot,  what version of image should we use now?

Reply:
Yes... you are right(but https://github.com/hstreamdb/hstream/blob/main/hstream/src/HStream/Server/Handler/Admin.hs#L462 is an admin command, not a normal describeConnector or listConnectors api).

Reply:
oh yes, I've confused them :(